lib/tools/src/xref_reader.erl

```diff
@@ -95,16 +96,19 @@ form({function, _, module_info, 0, _Clauses}, S) ->
     S;
 form({function, _, module_info, 1, _Clauses}, S) ->
     S;
-form({function, 0 = _Line, behaviour_info, 1, _Clauses}, S) ->
-    S;
 form({function, Anno, Name, Arity, Clauses}, S) ->
-    MFA0 = {S#xrefr.module, Name, Arity},
-    MFA = adjust_arity(S, MFA0),
-    S1 = S#xrefr{function = MFA},
-    Line = erl_anno:line(Anno),
-    S2 = S1#xrefr{def_at = [{MFA,Line} | S#xrefr.def_at]},
-    S3 = clauses(Clauses, S2),
-    S3#xrefr{function = []};
+    case {Name, Arity, erl_anno:location(Anno)} of
+        {behaviour_info, 1, 0} ->
```

Comment:
If I get `erl_internal.erl` correctly, `location` is hard-coded as `0`, here: `[{function,0,behaviour_info,1,`.
I'm testing this version (using `:location`), since I don't understand how it could cause a difference in behaviour (but probably there's some other piece of the puzzle I'm missing).

Reply:
Tested it and it's working as intended, in both the shell-based `xref` and the code-based `xref`.

Reply:
The change in xref_reader is no big deal. Matching an annotation with the integer zero breaks the erl_anno abstraction, that's all.